packages/fxa-settings/src/components/UnitRowSecondaryEmail/index.test.tsx

```diff
@@ -0,0 +1,237 @@
+          email: 'johndope@example.com',
+          isPrimary: true,
+          verified: true,
+        },
+        {
+          email: 'johndope2@example.com',
+          isPrimary: false,
+          verified: true,
+        },
+      ];
+      render(
+        <MockedCache account={{ emails }}>
+          <UnitRowSecondaryEmail />
+        </MockedCache>
+      );
+
+      expect(
+        screen.queryByTestId('resend-secondary-email-code-button')
+      ).not.toBeInTheDocument();
+      expect(screen.queryByTestId('unverified-text')).not.toBeInTheDocument();
+      expect(
+        screen.getByTestId('secondary-email-default-content')
+      ).toBeInTheDocument();
+      expect(
+        screen.getByTestId('secondary-email-make-primary')
+      ).toBeInTheDocument();
+    });
+  });
+
+  describe('multiple secondary emails set', () => {
+    it('renders as expected and with verified email text only present on the last verified', () => {
+      const emails = [
+        {
+          email: 'johndope@example.com',
+          isPrimary: true,
+          verified: true,
+        },
+        {
+          email: 'johndope2@example.com',
+          isPrimary: false,
+          verified: true,
+        },
+        {
+          email: 'johndope3@example.com',
+          isPrimary: false,
+          verified: true,
+        },
+        {
+          email: 'johndope4@example.com',
+          isPrimary: false,
+          verified: true,
+        },
+      ];
+      render(
+        <MockedCache account={{ emails }}>
+          <UnitRowSecondaryEmail />
+        </MockedCache>
+      );
+      const unitRowContents = screen.getAllByTestId('unit-row-content');
+      const unitRowHeaderValues = screen.getAllByTestId(
+        'unit-row-header-value'
+      );
+      const secondaryEmails = emails.filter((email) => !email.isPrimary);
+
+      expect(
+        screen.getAllByTestId('secondary-email-make-primary')
+      ).toHaveLength(3);
+      expect(
+        screen.getAllByTestId('secondary-email-default-content')
+      ).toHaveLength(1);
+
+      expect(unitRowContents[unitRowContents.length - 1]).toContainElement(
+        screen.getByTestId('secondary-email-default-content')
+      );
+
+      expect(unitRowHeaderValues).toHaveLength(3);
+      secondaryEmails.forEach((email, index) => {
+        expect(unitRowHeaderValues[index].textContent).toContain(email.email);
+      });
+    });
+
+    it('renders multiple unverified as expected', () => {
+      const emails = [
+        {
+          email: 'johndope@example.com',
+          isPrimary: true,
+          verified: true,
+        },
+        {
+          email: 'johndope2@example.com',
+          isPrimary: false,
+          verified: false,
+        },
+        {
+          email: 'johndope3@example.com',
+          isPrimary: false,
+          verified: true,
+        },
+        {
+          email: 'johndope4@example.com',
+          isPrimary: false,
+          verified: false,
+        },
```

Comment:
Heh, there are about 30 instances of this type of object. Maybe we can dry this up with a factory in `_mocks` or another shared test file? Something like:

```javascript
function mockAccountEmail(email, isPrimary, verified) {
  const email = email || autoGeneratedEmail
  return {
    email,
    isPrimary,
    verified,
  },
}
```

Reply:
Per Slack convo, I'm going to do this in a follow up and probably create a test utils or helpers file, it'll be paired with some doc updates.